mlir/include/mlir/Dialect/Affine/Analysis/LoopAnalysis.h

```diff
@@ -43,6 +43,10 @@ void getTripCountMapAndOperands(AffineForOp forOp, AffineMap *map,
 /// constant trip count in non-trivial cases.
 std::optional<uint64_t> getConstantTripCount(AffineForOp forOp);
 
+/// In some scenarios, such as GPU, the number of trip of each thread in the
+/// loop is inconsistent. This function returns the maximum number of trip.
```

Comment:
A doc comment should describe the utility generically instead of starting a description specific scenarios. Start doc comment with  "Returns ....". 

`getUpperBoundOnTripCount` appears to be a clearer name.